### PR TITLE
Keep timestamps when copy files

### PR DESCRIPTION
**Pull request by zprogd**
For some native libraries we have to keep timestamps of files from
tar archives. If change timestamp, 'make' will trigger full rebuild.
Normal workflow for libraries is unpack archive and build it.

We are changing timestamps during dircopy() when coping files to
build directory. I didn't find out how to force dircopy() to protect
timestamps.

One of solutions to avoid full rebuild is to touch some files:
$ touch aclocal.m4 configure Makefile.am Makefile.in

Another approach is don't change timestamps at all.

https://stackoverflow.com/questions/33278928/how-to-overcome-aclocal-1-15-is-missing-on-your-system-warning-when-compilin

### CODE REVIEW

**Comment by zprogd** [comment]
@mickeyn I don't like touch(). It can work for native/pixman, but it doesn't sound as universal solution, it may not work for some other libraries.

**Comment by mickeyn** [comment]
I think running a shell command is a better solution than using the module's methods.

**Comment by mickeyn** [comment]
also, if you're looking for universal solution - relying on specific shell command and its flags it not the way to go.

**Comment by gonzus** [comment]
Ideally, you would want dircopy() to support a flag similar to -p for cp:

    -p     same as --preserve=mode,ownership,timestamps

**Comment by xsawyerx** [comment]
`local $File::Copy::Recursive::KeepMode = 1;` should fix it.

**Comment by zprogd** [comment]
@xsawyerx It doesn't work. Seems that $File::Copy::Recursive::KeepMode works only for chmod https://metacpan.org/source/DMUEY/File-Copy-Recursive-0.38/Recursive.pm#L158

Should we add the same code for mtime in File-Copy-Recursive? But it is not our module and probably it may take some time to merge the patch.

**Comment by zprogd** [comment]
I rewrote the function. Can you guys review this code again?

**Comment by xsawyerx** [comment]
I suggest implementing the recursion with `Path::Tiny`.

**Comment by mickeyn** [comment]
I don't see any reason to change the access time. 
Changing the modification time can still be achieved with Path::Tiny's touch(epoch).
I agree with @xsawyerx re recurse.

**Comment by zprogd** [comment]
@mickeyn @xsawyerx : done

File: lib/Pakket/Builder.pm [hunk 1]
```diff
@@ -436,6 +441,19 @@ sub run_build {
     return;
 }
 
+sub fix_timestamps {
+    my ($src, $dst) = @_;
+    $src->visit(
+        sub {
+            my $path = shift;
+            my $mtime = $path->stat->[9];
```
Comment (mickeyn) [inline lib/Pakket/Builder.pm]: s/ \\[ 9 \\ ] /mtime/x

**Pull request merged**